presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1769,15 +1769,20 @@ private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progres
         protected Iterable<List<Object>> computeNext()
         {
             while (client.isRunning()) {
-                if (Thread.currentThread().isInterrupted()) {
-                    client.close();
-                    throw new RuntimeException(new SQLException("ResultSet thread was interrupted"));
-                }
+                checkInterruption();
 
                 QueryStatusInfo results = client.currentStatusInfo();
                 progressCallback.accept(QueryStats.create(results.getId(), results.getStats()));
                 Iterable<List<Object>> data = client.currentData().getData();
-                client.advance();
+
+                try {
+                    client.advance();
+                }
+                catch (RuntimeException e) {
+                    checkInterruption();
+                    throw e;
```

Comment:
`e` gets lost if `checkInterruption` throws. This may or may not be the interruption exception, so can be potentially interesting. Add to suppressed?